src/dc_receive_imf.rs

```diff
@@ -60,673 +59,613 @@ pub unsafe fn dc_receive_imf(
+    if mime_parser.header.is_empty() {
+        info!(context, 0, "No header.",);
     } else {
-        dc_mimeparser_parse(&mut mime_parser, imf_raw_not_terminated, imf_raw_bytes);
-        if mime_parser.header.is_empty() {
-            dc_log_info(
-                context,
-                0i32,
-                b"No header.\x00" as *const u8 as *const libc::c_char,
-            );
-        } else {
-            /* Error - even adding an empty record won't help as we do not know the message ID */
-            field = dc_mimeparser_lookup_field(
-                &mut mime_parser,
-                b"Date\x00" as *const u8 as *const libc::c_char,
-            );
-            if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_ORIG_DATE as libc::c_int {
-                let orig_date: *mut mailimf_orig_date = (*field).fld_data.fld_orig_date;
-                if !orig_date.is_null() {
-                    sent_timestamp = dc_timestamp_from_date((*orig_date).dt_date_time)
-                }
+        /* Error - even adding an empty record won't help as we do not know the message ID */
+        field = dc_mimeparser_lookup_field(&mut mime_parser, "Date");
+        if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_ORIG_DATE as libc::c_int {
+            let orig_date: *mut mailimf_orig_date = (*field).fld_data.fld_orig_date;
+            if !orig_date.is_null() {
+                sent_timestamp = dc_timestamp_from_date((*orig_date).dt_date_time)
             }
-            field = dc_mimeparser_lookup_field(
-                &mime_parser,
-                b"From\x00" as *const u8 as *const libc::c_char,
-            );
-            if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_FROM as libc::c_int {
-                let fld_from: *mut mailimf_from = (*field).fld_data.fld_from;
-                if !fld_from.is_null() {
-                    let mut check_self: libc::c_int = 0;
-                    let from_list: *mut dc_array_t = dc_array_new(16i32 as size_t);
-                    dc_add_or_lookup_contacts_by_mailbox_list(
-                        context,
-                        (*fld_from).frm_mb_list,
-                        0x10i32,
-                        from_list,
-                        &mut check_self,
-                    );
-                    if 0 != check_self {
-                        incoming = 0i32;
-                        if 0 != dc_mimeparser_sender_equals_recipient(&mime_parser) {
-                            from_id = 1i32 as uint32_t
-                        }
-                    } else if dc_array_get_cnt(from_list) >= 1 {
-                        from_id = dc_array_get_id(from_list, 0i32 as size_t);
-                        incoming_origin =
-                            dc_get_contact_origin(context, from_id, &mut from_id_blocked)
+        }
+        field = dc_mimeparser_lookup_field(&mime_parser, "From");
+        if !field.is_null() && (*field).fld_type == MAILIMF_FIELD_FROM as libc::c_int {
```

Comment:
Doesn't rust figure out that `field.fld_type` is really `(*field).fld_type`?  not that it does any harm